peer.go

```diff
@@ -425,6 +425,45 @@ func (p *peer) Start() error {
 		}
 	}
 
+	// Node announcements don't propagate very well throughout the network
+	// as there isn't a way to efficiently query for them through their
+	// timestamp, mostly affecting nodes that were offline during the time
+	// of broadcast. We'll resend our node announcement to the remote peer
+	// as a best-effort delivery such that it can also propagate to their
+	// peers.  To ensure they can successfully process it in most cases,
+	// we'll only resend it as long as we have at least one confirmed
+	// advertised channel.
+	//
+	// TODO(wilmer): Remove this once we're able to query for node
+	// announcements through their timestamps.
+	hasConfirmedPublicChan := false
+	for _, activeChan := range activeChans {
+		if activeChan.IsPending {
+			continue
+		}
+		if activeChan.ChannelFlags&lnwire.FFAnnounceChannel == 0 {
+			continue
+		}
+
+		hasConfirmedPublicChan = true
+		break
+	}
+
+	if hasConfirmedPublicChan {
```

Comment:
so now we'll only send our node if we have a public, confirmed channel with the target peer, i think this is a good compromise. it's not perfect, but is definitely a big improvement over the status quo